theme/views.py

```diff
@@ -30,11 +31,23 @@ def get_context_data(self, **kwargs):
             except:
                 u = User.objects.get(username=self.request.GET['user'])
 
-        res = BaseResource.objects.filter(user=u)
+        # get all resources the profile user owns
+        resources = u.uaccess.get_owned_resources()
+
+        # if requesting user is not the profile user, then show only resources that the requesting user has access
+        if self.request.user != u:
+            if self.request.user.is_authenticated():
+                # filter out any resources the requesting user doesn't have access
+                resources = resources.filter(Q(pk__in=self.request.user.uaccess.get_held_resources()) |
```

Comment:
@pkdash @ironfroggy I am wondering whether there is more efficient way to do filtering here. The current logic calls filtering on BaseResource twice, once from get_owned_resources(), and once from get_held_resources() with the purpose to getting the intersection of these two querysets.  If there is no more efficient way to do filtering here, I am ok with this change.